test/libyul/ObjectParser.cpp

```diff
@@ -258,16 +258,6 @@ BOOST_AUTO_TEST_CASE(arg_to_dataoffset_must_be_literal)
 	CHECK_ERROR(code, TypeError, "Function expects direct literals as arguments.");
 }
 
-BOOST_AUTO_TEST_CASE(arg_to_dataoffset_must_be_string_literal)
```

Comment:
Why removing these tests?

Reply:
They're in https://github.com/ethereum/solidity/pull/9569/files#diff-a4bb2c1dafc8bc4ce2cac5b1b997ae3f right? And I was too lazy to fix the expectation in the cpp file manually :-).

Reply:
They are but the comprehensive object test in this file, until we get #9502 finished and merged :)

Reply:
It doesn't matter too much where those tests are, as long as they're there, does it :-)?
And at the latest *given* https://github.com/ethereum/solidity/pull/9502 we can remove these here all the more :-). But if you insist, I'll add it back and fix the expectation instead...

Reply:
And the tests in this file aren't comprehensive anyhow, since they don't test ``setimmutable`` and ``loadimmutable`` (which are also specifically part of the object parser), right :-)?

Reply:
If you insist removing them I'd rather remove it in the commit adding it, and no adding and removing it within the same PR...

Reply:
I'm fine either way... I'd actually squash the whole PR into a single commit anyways, though...

Reply:
Actually, the reason for the mess in #9502 is that `yulSyntaxTests` don't go through `libyul/ObjectParser`, while `test/libyul/ObjectParser.cpp` does.

Reply:
Well, the way the PR works now, those errors are not related to the object parser, but are rather builtin-related analyzer errors, though, aren't they? But yeah, I really don't mind, but if I see it correctly, you now rebased any trace of these tests away anyways, so this is good now :-)?

Reply:
Ah, no, just looked like it, because indeed they are added by one commit and removed by the other... well, in any case, I'd personally just squash the whole thing. But you can also add the tests back here.

Reply:
Squashed more, ready for review.

Reply:
Any reason to not just squash it entirely?

Reply:
(Or at least the first three commits?)